src/app/modules/navigator/test-exec-navigator.component.spec.ts

```diff
@@ -329,4 +329,91 @@ describe('TestExecNavigatorComponent', () => {
     // TODO: test to come as soon as test are polled and the end of a test run is actually detected
   });
 
+  it('constructs a (readable) entry for test executions', () => {
+    // given
+    const tclPath = 'src/test/java/some/Test.tcl';
+
+    // when
+    const testRun = component.buildUITestRunFromSingleTest(tclPath, 'someUrl', new Date('1971-01-28 15:30'));
+
+    // then
+    expect(testRun.name).toEqual('15:30 Test (some)');
+    expect(testRun.paths).toEqual([tclPath]);
+    expect(testRun.url).toEqual('someUrl');
+    expect(testRun.class).toEqual('running');
+    // jasmine.objectContaining does not work here, since testRun.class is of type UITestRunStatus
+    // (which essentially is of type string, but type conformance calculation seems to fail here)
+  });
+
+  it('shows no test run button if no test runs are available', () => {
+    // given + when +then
+    const testRunButtons = fixture.debugElement.queryAll(By.css('.testRunButton'));
+
+    expect(testRunButtons).toEqual([]);
+  });
+
+  it('shows the test run button if test runs are available', () => {
+    // given
+    const tclPath = 'src/test/java/some/Test.tcl';
+
+    // when
+    const testRun = component.buildUITestRunFromSingleTest(tclPath, 'someUrl', new Date('1971-01-28 15:30'));
+    component.addTestRun(testRun);
+    fixture.detectChanges();
+
+    // then
+    const testRunButton = fixture.debugElement.queryAll(By.css('.testRunButton'))[0].nativeElement;
+    expect(testRunButton.disabled).toBeFalsy();
+  });
+
+  it('loads a test if selecting it from the test run dropdown', () => {
+    // given
+    const tclPath = 'src/test/java/some/Test.tcl';
+    spyOn(component, 'loadExecutedTreeFor');
+    when(testExecutionServiceMock.getCallTree('someUrl')).thenResolve({
+      testSuiteId: '0',
+      testSuiteRunId: '0',
+      resourcePaths: [ tclPath ],
+      testRuns: []
+    });
+
+    // when
+    const testRun = component.buildUITestRunFromSingleTest(tclPath, 'someUrl', new Date('1971-01-28 15:30'));
+    component.addTestRun(testRun);
+    fixture.detectChanges();
+    const testRunDropdownElement = fixture.debugElement.queryAll(By.css('.testRunDropdown li:first-child'));
+    testRunDropdownElement[0].nativeElement.click();
+    fixture.detectChanges();
+
+    // then
+    expect(component.loadExecutedTreeFor).toHaveBeenCalledWith(tclPath, 'someUrl');
+  });
+
+  // this test runs for chrome only
```

Comment:
Is this still the case or can the test be re-enabled?

Reply:
i cannot make it work for the firefox, no matter what i tried, it simply won't budge. since it does work for chrome I still think it might be useful.